message_buffer/programs/mock-cpi-caller/tests/program_test/mod.rs

```diff
@@ -0,0 +1,681 @@
+    let version = cursor.read_u8().unwrap();
+    let header_len = cursor.read_u16::<LittleEndian>().unwrap();
+    let mut end_offsets = [0u16; 255];
+    for i in 0..255 {
+        let cur_end_offset = cursor.read_u16::<LittleEndian>().unwrap();
+        end_offsets[i] = cur_end_offset;
+    }
+
+    let mut messages = vec![];
+    cursor.read_to_end(&mut messages).unwrap();
+
+    (msg_buffer_acct_bump, version, header_len, end_offsets)
+}
+
+pub fn extract_msg_buffer_messages(
+    header_len: u16,
+    end_offsets: EndOffsets,
+    account_data: &[u8],
+) -> Vec<Vec<u8>> {
+    let mut msgs = vec![];
+    let mut msg_begin = header_len;
+    for end_offset in end_offsets {
+        if end_offset == 0 {
+            break;
+        }
+        let msg_end = header_len + end_offset;
+        msgs.push(account_data[(msg_begin as usize)..(msg_end as usize)].to_vec());
+        msg_begin = msg_end;
+    }
+    msgs
+}
+
+pub fn sighash(namespace: &str, name: &str) -> [u8; 8] {
+    let preimage = format!("{namespace}:{name}");
+
+    let mut sighash = [0u8; 8];
+    sighash.copy_from_slice(&hashv(&[preimage.as_bytes()]).to_bytes()[..8]);
+    sighash
+}
+
+// price::MessageHeader
+type PriceMsgSchema = u8;
+type PriceMsgVersion = u16;
+type PriceMsgSize = u32;
+
+
+pub fn validate_price_msgs(
+    id: u64,
+    price: u64,
+    price_expo: u64,
+    ema: u64,
+    ema_expo: u64,
+    msgs: &Vec<Vec<u8>>,
+) -> Result<()> {
+    for msg in msgs {
+        let (schema, _, _) = deserialize_price_msg_header(msg);
+        match schema {
+            0 => verify_full_price_msg(id, price, price_expo, ema, ema_expo, msg)?,
+            1 => verify_compact_price_msg(id, price, price_expo, msg)?,
+            _ => panic!(),
+        }
+    }
+    Ok(())
+}
+
+pub fn verify_full_price_msg(
+    id: u64,
+    price: u64,
+    price_expo: u64,
+    ema: u64,
+    ema_expo: u64,
+    msg: &[u8],
+) -> Result<()> {
+    let mut cursor = Cursor::new(&msg[7..]);
+    let msg_id = cursor.read_u64::<BigEndian>().unwrap();
+    assert_eq!(id, msg_id);
+    let msg_price = cursor.read_u64::<BigEndian>().unwrap();
+    assert_eq!(price, msg_price);
+    let msg_price_expo = cursor.read_u64::<BigEndian>().unwrap();
+    assert_eq!(price_expo, msg_price_expo);
+    let msg_ema = cursor.read_u64::<BigEndian>().unwrap();
+    assert_eq!(ema, msg_ema);
+    let msg_ema_expo = cursor.read_u64::<BigEndian>().unwrap();
+    assert_eq!(ema_expo, msg_ema_expo);
+    Ok(())
+}
+
+pub fn verify_compact_price_msg(id: u64, price: u64, price_expo: u64, msg: &[u8]) -> Result<()> {
+    let mut cursor = Cursor::new(&msg[7..]);
+    let msg_id = cursor.read_u64::<BigEndian>().unwrap();
+    assert_eq!(id, msg_id);
+    let msg_price = cursor.read_u64::<BigEndian>().unwrap();
+    assert_eq!(price, msg_price);
+    let msg_price_expo = cursor.read_u64::<BigEndian>().unwrap();
+    assert_eq!(price_expo, msg_price_expo);
+    Ok(())
+}
```

Comment:
I think it's good if we can remove them. I don't have the full context on the code. Is there any reason that we are manually ser/de ing on the mock contract? I think we can use borsh (and it will make the code simpler)

Reply:
@Reisen was working on a serializer/deserializer for the pythnet sdk. This was just to have some code to work off of that didn't rely on using the anchor client libraries